src/Microsoft.OpenApi/Reader/OpenApiReaderRegistry.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.OpenApi.Interfaces;
+
+namespace Microsoft.OpenApi.Reader
+{
+    /// <summary>
+    /// Registry for managing different OpenAPI format providers.
+    /// </summary>
+    public static class OpenApiReaderRegistry
+    {
+        private static readonly Dictionary<string, IOpenApiReader> _readers = new(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Defines a default OpenAPI reader.
+        /// </summary>
+        public static readonly IOpenApiReader DefaultReader = new OpenApiJsonReader();
+
+        /// <summary>
+        /// Registers an IOpenApiReader for a given OpenAPI format.
+        /// </summary>
+        /// <param name="format">The OpenApi file format.</param>
+        /// <param name="reader">The reader instance.</param>
+        public static void RegisterReader(string format, IOpenApiReader reader)
+        {
+            _readers[format] = reader;
```

Comment:
consider normalizing the format.
ToLower() or something.

Reply:
This won't be necessary since I've initialized the _readers dictionary with a case-insensitive string comparer hence all entries are treated as case-insensitive so we don't need to normalize the format